### ISSUE TITLE
Missing routeWillChange & routeDidChange

### ISSUE THREAD

**Issue by nickiaconis** [rwjblue/ember-router-service-polyfill#1]
The polyfill presently lacks support for the `routeWillChange` and `routeDidChange` events. They are described [here in emberjs/rfcs#95](https://github.com/emberjs/rfcs/blob/master/text/0095-router-service.md#new-events-routewillchange--routedidchange). This, at the very least, prevents ember-prefetch from removing private API monkey patches in favor of this polyfill.

**Comment by rwjblue** [comment]
Yep, this polyfill only currently provides access to methods/properties that do not require the RouteInfo objects (as that work is not yet done).

We should update the README's known limitations section to mention what we haven't implemented yet.

**Comment by mansona** [comment]
Hey folks 👋 is there any appetite for someone to do this work? I'm wondering if there is a plan for how you would be able to achieve this as a polyfill and if so then I might be able to take a quick look at it 😄

**Comment by rwjblue** [comment]
Seems fine to do, but at this point the value is pretty low. There are no supported Ember LTS's that don't already have these events...

**Comment by mansona** [comment]
yes that may be true, but it could help people on older versions of Ember on their upgrade path. This came up for an addon that I wanted to upgrade to the new paradigm but didn't want to break backwards compatibility.

Obviously there is **limited** value but I'm wondering what your opinion on the expected effort might be 🤔  I am happy to dive into this but I haven't had a look at what it would take quite yet
